File: src/python/finished/binary_prefix_divisible_by_5.py
from typing import List

# https://leetcode.com/problems/binary-prefix-divisible-by-5/


class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        n = len(nums)
        ans = [False] * n
        cur = 0
        for i, n in enumerate(nums):
            cur = (cur << 1) + n
            cur = cur % 5
            ans[i] = cur == 0
        return ans

File: src/rust/finished/binary_prefix_divisible_by_5.rs
impl Solution {
    pub fn prefixes_div_by5(nums: Vec<i32>) -> Vec<bool> {
        let mut acc = 0;
        let mut ans = Vec::with_capacity(nums.len());
        for n in nums.into_iter() {
            acc = (acc << 1) + n;
            acc = acc.rem_euclid(5);
            ans.push(acc == 0)
        }
        ans
    }
}
